wagtailstreamforms/forms.py

```diff
@@ -38,18 +38,9 @@ def formfields(self):
                     "Could not find a registered field of type %s" % field_type
                 )
 
-            # check there is a label
-            if "label" not in field_value:
-                raise AttributeError(
-                    "The block for %s must contain a label of type blocks.CharBlock(required=True)"
-                    % field_type
-                )
-
-            # slugify the label for the field name
-            field_name = get_slug_from_string(field_value.get("label"))
-
             # get the field
             registered_cls = registered_fields[field_type]()
+            field_name = registered_cls.get_formfield_name(field_value)
```

Comment:
I love it. I made a similar change in https://github.com/rgs258/wagtailstreamforms/pull/1/files#diff-951d949d800424b6334469000bf4a47f1841b3fe5b59b25b81d07a779d9a3c11R55